### PR TITLE
Emit disk metrics with paasta information

### PR DESCRIPTION
**Pull request by JayatiTiwari**
Fixed the code to incorporate the review comments.

### CODE REVIEW

File: src/fullerite/collector/docker_stats.go [hunk 1]
```diff
@@ -129,6 +170,20 @@ func (d *DockerStats) Collect() {
...
 	}
+
+	if d.emitDiskMetrics {
+		// Obtain the disk stats for this device. This is common for all the containers, hence, calculating before iterating over all the containers.
+		diskStats, err = d.ObtainDiskStats()
+		if err != nil {
+			d.log.Error("ObtainDiskStats() failed: ", err)
+		}
+		// join the disk stats with the IO stats
+		diskIOStatsList, err = d.ObtainDiskIOStats(diskStats)
```
Comment (flavray) [inline src/fullerite/collector/docker_stats.go]: If `d.ObtainDiskIOStats` above returns an error, `diskStats` may be `nil` and cause this function to panic. Would it make sense to either: not call this function if `err != nil`, or make `diskStats = make(map[string][]string)` in the error handling above?

File: src/fullerite/collector/docker_stats.go [hunk 2]
```diff
@@ -308,3 +391,172 @@ func min(x, y int) int {
...
+						}
+					}
+				}
+			}
+		}
+	}
+	return paastaIOStatsList
+}
+
+func ObtainRealPath(mountPath string) (string, error) {
```
Comment (flavray) [inline src/fullerite/collector/docker_stats.go]: nice! 🙂

File: src/fullerite/collector/docker_stats.go [hunk 3]
```diff
@@ -308,3 +391,172 @@ func min(x, y int) int {
...
+	mountPathReal, err := realpath.Realpath(mountPath)
+	if err == nil {
+		return mountPathReal, nil
+	}
+	return mountPath, err
```
Comment (flavray) [inline src/fullerite/collector/docker_stats.go]: nit: should we `return nil, err` here?

File: src/fullerite/collector/docker_stats.go [hunk 4]
```diff
@@ -308,3 +391,172 @@ func min(x, y int) int {
...
+	// read the contents of the file with a reader.
+	reader := bufio.NewReader(file)
+
+	// read line-by-line
+	var line string
+	for {
+		line, err = reader.ReadString('\n')
+
+		if err != nil {
+			break
```
Comment (flavray) [inline src/fullerite/collector/docker_stats.go]: this looks like the only break condition in the loop, doesn't it mean `err` will never be `nil` when returning (Line 438)? If so, won't this call technically always fail?

File: src/fullerite/collector/docker_stats.go [hunk 5]
```diff
@@ -308,3 +391,172 @@ func min(x, y int) int {
...
+			if !strings.HasPrefix(rec[2], "ram") && !strings.HasPrefix(rec[2], "loop") && major == 202 {
+				devNameMinMajMap[rec[2]] = []string{rec[0], rec[1], rec[2], rec[3], rec[4], rec[5], rec[6], rec[7], rec[8], rec[9], rec[10], rec[11], rec[12], rec[13]}
+			}
+		} else {
+			d.log.Warning("This record of /proc/diskstats does not contain the required number of fields: " + line + " Shall not be processed.")
+		}
+
+		if err != io.EOF {
+			d.log.Error("Failed!: ", err)
+		}
```
Comment (flavray) [inline src/fullerite/collector/docker_stats.go]: Should this code be moved out of the loop?

File: src/fullerite/collector/docker_stats.go [hunk 6]
```diff
@@ -308,3 +391,172 @@ func min(x, y int) int {
...
+	// read contents of the file
+	reader := bufio.NewReader(file)
+
+	// read line-by-line
+	var line string
+	for {
+		line, err = reader.ReadString('\n')
+
+		if err != nil {
+			break
```
Comment (flavray) [inline src/fullerite/collector/docker_stats.go]: See above

File: src/fullerite/collector/docker_stats.go (hunk 1 above)
Comment (JayatiTiwari) [reply src/fullerite/collector/docker_stats.go]: Thanks. But, we are making it on line 161 they way you mentioned :)

        diskStats := make(map[string][]string)

File: src/fullerite/collector/docker_stats.go (hunk 2 above)
Comment (JayatiTiwari) [reply src/fullerite/collector/docker_stats.go]: :)

File: src/fullerite/collector/docker_stats.go (hunk 4 above)
Comment (JayatiTiwari) [reply src/fullerite/collector/docker_stats.go]: Resolved by:
// outside the for loop
if err != io.EOF {
                        d.log.Error("Failed!: ", err)
                        return nil, err
 }
return devNameMinMajMap, nil

File: src/fullerite/collector/docker_stats.go (hunk 5 above)
Comment (JayatiTiwari) [reply src/fullerite/collector/docker_stats.go]: Done. Thanks

File: src/fullerite/collector/docker_stats.go (hunk 6 above)
Comment (JayatiTiwari) [reply src/fullerite/collector/docker_stats.go]: Resolved the same way.

File: src/fullerite/collector/docker_stats.go (hunk 3 above)
Comment (JayatiTiwari) [reply src/fullerite/collector/docker_stats.go]: Got the below on using nil:

src/fullerite/collector/docker_stats.go:561:3: cannot use nil as type string in return argument

hence used "" 

Reference:
https://stackoverflow.com/questions/12703243/what-is-the-zero-for-string/12703280

**Pull request merged**